api/deploy.go

```diff
@@ -461,9 +464,17 @@ func deployRollbackUpdate(w http.ResponseWriter, r *http.Request, t auth.Token)
 	defer func() { evt.Done(err) }()
 	err = app.RollbackUpdate(instance, img, reason, rollback)
 	if err != nil {
-		return &tsuruErrors.HTTP{
-			Code:    http.StatusConflict,
-			Message: err.Error(),
+		switch err {
+		case image.ErrInvalidVersion:
+			return &tsuruErrors.HTTP{
+				Code:    http.StatusInternalServerError,
```

Comment:
I think the errors here are backwards: first case should be `http.StatusBadRequest`. You actually do not need the second case (you can just return the error directly and it will be represented as a ` http.StatusInternalServerError` by our [error handling middleware](https://github.com/tsuru/tsuru/blob/456a675a3aba0163195d66c14688f9a59c2c7b6c/api/middleware.go#L102).